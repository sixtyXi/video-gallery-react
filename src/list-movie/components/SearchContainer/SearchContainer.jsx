import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import queryString from 'query-string';
import PropTypes from 'prop-types';

import styles from './SearchContainer.scss';

import SearchBtnContainer from '../SearchBtnContainer/SearchBtnContainer';
import SearchFieldContainer from '../SearchFieldContainer/SearchFieldContainer';
import SearchFilterContainer from '../SearchFilterContainer/SearchFilterContainer';

export class SearchContainer extends Component {
  pushSearchUrl = () => {
    const { search, searchBy, sortBy, history } = this.props;

    history.push(`/search?${queryString.stringify({ search, searchBy, sortBy })}`);
  };

  render() {
    return (
      <form className={styles.searchWrapper}>
        <h1 className={styles.pageTitle}>Find your movie</h1>

        <SearchFieldContainer pushUrl={this.pushSearchUrl} />

        <SearchFilterContainer />

        <SearchBtnContainer pushUrl={this.pushSearchUrl} />
      </form>
    );
  }
}

SearchContainer.propTypes = {
  search: PropTypes.string.isRequired,
  searchBy: PropTypes.string.isRequired,
  sortBy: PropTypes.string.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired
};

const mapStateToProps = state => ({
  search: state.movieList.search,
  searchBy: state.movieList.searchBy,
  sortBy: state.movieList.sortBy
});

export default withRouter(connect(mapStateToProps)(SearchContainer));
