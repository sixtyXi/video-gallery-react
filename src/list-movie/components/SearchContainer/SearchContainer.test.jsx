import React from 'react';
import { shallow } from 'enzyme';

import { SearchContainer } from './SearchContainer';

describe('SearchContainer', () => {
  test('renders correctly', () => {
    const wrapper = shallow(<SearchContainer />);

    expect(wrapper).toMatchSnapshot();
  });
});
