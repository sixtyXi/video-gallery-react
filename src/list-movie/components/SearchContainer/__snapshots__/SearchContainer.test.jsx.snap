// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchContainer renders correctly 1`] = `
<form
  className="searchWrapper"
>
  <h1
    className="pageTitle"
  >
    Find your movie
  </h1>
  <withRouter(Connect(SearchFieldContainer))
    pushUrl={[Function]}
  />
  <Connect(SearchFilterContainer) />
  <Connect(SearchBtnContainer)
    pushUrl={[Function]}
  />
</form>
`;
